<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <!--设置目标距离和角度-->
  <arg name="target_distance" default="1.0" />
  <arg name="target_angle" default="0" />

  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen">
    <param name="video_device" value="/dev/video2" />
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="pixel_format" value="yuyv" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap"/>
  </node>

  <node name='laser_to_camera' pkg="simu_follower" type="laseroncamera.py" output="screen"/>
  <!--根据距离角度设置cmd_vel消息中的值-->
  <node name="follower" pkg="simu_follower" type="follower.py" output="screen">
    <param name="maxSpeed" value="2.0" type="double" />
    <param name="maxAngleSpeed" value="2.0" type="double" />
    <param name="targetDis" value="$(arg target_distance)" type="double" />
    <param name="targetAngle" value="$(arg target_angle)" type="double" />
    <rosparam ns="PID_controller" command="load" file="$(find simu_follower)/parameters/PID_param.yaml" />
  </node>
  <!--node name="image_view" pkg="image_view" type="image_view" respawn="false">
    <remap from="image" to="/usb_cam/image_raw"/>
    <param name="autosize" value="true" />
  </node-->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find simu_follower)/rviz/laserONcam.rviz" />
</launch>
